bundles/persistence/org.openhab.persistence.sitewhere/src/main/java/org/openhab/persistence/sitewhere/internal/SiteWherePersistenceService.java

```diff
@@ -0,0 +1,443 @@
+	protected ItemRegistry itemRegistry;
+
+	/** Event publisher for openHAB */
+	private EventPublisher eventPublisher;
+
+	/** Indicates if the service has been configured */
+	protected boolean configured = false;
+
+	/** Indicates if REST connectivity has been verified */
+	protected boolean restVerified = false;
+
+	/** Indicates if SiteWhere agent is connected */
+	protected boolean agentConnected = false;
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see org.openhab.core.persistence.PersistenceService#getName()
+	 */
+	public String getName() {
+		return SERVICE_NAME;
+	}
+
+	/**
+	 * OSGI setter for {@link EventPublisher}.
+	 * 
+	 * @param publisher
+	 */
+	public void setEventPublisher(EventPublisher publisher) {
+		this.eventPublisher = publisher;
+	}
+
+	/**
+	 * OSGI unsetter for {@link EventPublisher}.
+	 * 
+	 * @param publisher
+	 */
+	public void unsetEventPublisher(EventPublisher publisher) {
+		this.eventPublisher = null;
+	}
+
+	/**
+	 * Method called when bundle is activated.
+	 */
+	public void activate() {
+		LOGGER.debug("Activate method called on persistence service.");
+		try {
+			SiteWhereConfigurationManager.getInstance();
+			setConfigured(true);
+			validateSiteWhereClient();
+			validateSiteWhereAgent();
+		} catch (Throwable t) {
+			LOGGER.error("SiteWhere activation failed.", t);
+			setConfigured(false);
+		}
+	}
+
+	/**
+	 * Method called when bundle is deactivated.
+	 */
+	public void deactivate() {
+		LOGGER.debug("Deactivate method called on persistence service.");
+		setConfigured(false);
+	}
+
+	/**
+	 * Set reference to item registry.
+	 * 
+	 * @param itemRegistry
+	 */
+	public void setItemRegistry(ItemRegistry itemRegistry) {
+		this.itemRegistry = itemRegistry;
+		LOGGER.debug("Item registry set for SiteWhere persistence.");
+	}
+
+	/**
+	 * Unset reference to item registry.
+	 * 
+	 * @param itemRegistry
+	 */
+	public void unsetItemRegistry(ItemRegistry itemRegistry) {
+		this.itemRegistry = null;
+	}
+
+	/**
+	 * Validate that client is able to interact with SiteWhere server.
+	 * 
+	 * @return
+	 */
+	protected boolean validateSiteWhereClient() {
+		try {
+			SiteWhereConfiguration config = SiteWhereConfigurationManager
+					.getInstance().getConfiguration();
+			this.client = new SiteWhereClient(config.getConnection()
+					.getRestApiUrl(), config.getConnection()
+					.getRestApiUsername(), config.getConnection()
+					.getRestApiPassword(), 2000);
+
+			Version version = client.getSiteWhereVersion();
+			LOGGER.info("Connected to SiteWhere "
+					+ version.getVersionIdentifier() + " "
+					+ version.getEditionIdentifier());
+			setRestVerified(true);
+			return true;
+		} catch (SiteWhereException e) {
+			this.client = null;
+			setRestVerified(false);
+			LOGGER.error("Unable to connect with SiteWhere server. Verify configuration.");
+			return false;
+		}
+	}
+
+	/**
+	 * Validate that the SiteWhere MQTT agent can connect.
+	 * 
+	 * @return
+	 */
+	protected boolean validateSiteWhereAgent() {
```

Comment:
since this is the method where the agent is created i would rename this method to ```createAgent``` or alike and explain other functionality in javadoc